### PR TITLE
Do not call get project.getBasePath

### PR DESCRIPTION
**Pull request by tpasternak**
This method may return incorrect value. The case is that the user may
create project literally *anywhere* and tnen import BSP project to
it. `getBasePath` will return path to the directory where the project
was created.

Apart from that, since this commit the `sources-jar` dir is also
recognized as project internal dependency (updated
`isProjectInternalDependency` method)

### CODE REVIEW

**Comment by tpasternak** [comment]
@lukaszwawrzyk

File: src/com/twitter/intellij/pants/bsp/JarMappings.java [hunk 1]
```diff
@@ -60,11 +60,14 @@ public void after(@NotNull List<? extends VFileEvent> events) {
...
-        VirtualFile file = librariesFile();
-        String content = new String(file.contentsToByteArray());
-        Type mapType = new TypeToken<Map<String, String>>() {}.getType();
-        libraryJarToLibrarySourceJar = new Gson().fromJson(content, mapType);
-        librariesFileIsUpToDate = true;
+        Optional<VirtualFile> file = librariesFile();
+        if(file.isPresent()) {
+          String content = new String(file.get().contentsToByteArray());
+          Type mapType = new TypeToken<Map<String, String>>() {
+          }.getType();
```
Comment (lukaszwawrzyk) [inline src/com/twitter/intellij/pants/bsp/JarMappings.java:67]: Maybe fix this awkward formatting ;p

File: src/com/twitter/intellij/pants/bsp/JarMappings.java [hunk 2]
```diff
@@ -125,17 +128,27 @@ private String targetAddressFromSanitizedFileName(String jarName) {
 
   private boolean isProjectInternalDependency(VirtualFile jar) {
     Path jarPath = Paths.get(jar.getPath());
-    Path bloopJarsPath = bloopJarsPath(project);
-    return jarPath.startsWith(bloopJarsPath);
+    Optional<Path> bloopJarsPath = bloopJarsPath(project);
+    Optional<Path> sourcesJarPath = sourcesJarPath(project);
+    return jarPath.startsWith(bloopJarsPath.get()) || jarPath.startsWith(sourcesJarPath.get());
```
Comment (lukaszwawrzyk) [inline src/com/twitter/intellij/pants/bsp/JarMappings.java:133]: are these `get` calls safe?

File: src/com/twitter/intellij/pants/bsp/JarMappings.java [hunk 3]
```diff
@@ -125,17 +128,27 @@ private String targetAddressFromSanitizedFileName(String jarName) {
...
+  private static Optional<Path> bloopJarsPath(Project project) {
+    return PantsBspData.importsFor(project).stream().findFirst()
+      .map(path -> Paths.get(path.getBspPath().toString(), ".bloop", "bloop-jars"));
   }
 
-  private VirtualFile librariesFile() {
-    Path path = Paths.get(project.getBasePath(), ".pants", "libraries.json");
-    return LocalFileSystem.getInstance().findFileByIoFile(path.toFile());
+  private static Optional<Path> sourcesJarPath(Project project) {
+    return PantsBspData.importsFor(project).stream().findFirst()
```
Comment (lukaszwawrzyk) [inline src/com/twitter/intellij/pants/bsp/JarMappings.java:142]: we could have a method to return bloop directory and then this and bloopJarsPath could just resolve single child

**Pull request merged**